src/main/java/core/Storage.java

```diff
@@ -0,0 +1,44 @@
+package core;
+
+import command.CommandHandler;
+import tasks.Task;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Storage {
+    /**
+     * Saves the task list to a file.
+     * @throws IOException if there was an error writing to the file
+     */
+    public static void save() throws IOException {
+        final File saveFile = new File("./data/saved.txt");
+        saveFile.getParentFile().mkdirs();
+        saveFile.createNewFile();
+        final FileWriter fw = new FileWriter(saveFile);
+        fw.write(IntStream.rangeClosed(1, TaskList.size())
+                .boxed()
+                .map(i ->  {
+                    Task t = TaskList.get(i - 1);
+                    return t.getParentCommand() + (t.getDoneStatus() ? "\ndone " + i : "");
```

Comment:
Perhaps you can break before the operator '+' to improve readability

Reply:
Ok, sounds reasonable since the second half represents a new line in the string.